File: algorithms/easy/apple_redistribution_into_boxes.rs
impl Solution {
    pub fn minimum_boxes(apple: Vec<i32>, mut capacity: Vec<i32>) -> i32 {
        let mut total: i32 = apple.iter().sum();
        capacity.sort_by(|a, b| b.cmp(a));
        for i in 0..capacity.len() {
            total -= capacity[i];
            if total <= 0 {
                return i as i32 + 1;
            }
        }
        
        unreachable!()
    }
}

File: algorithms/easy/apple_redistribution_into_boxes.kt
class Solution {
  fun minimumBoxes(apple: IntArray, capacity: IntArray): Int {
    var totalApples = apple.sum()
    var ct = 0

    capacity.sortDescending()
    for (c in capacity) {
      totalApples -= c
      ct++
      if (totalApples <= 0) {
        break
      }
    }

    return ct
  }
}
